hs_core/hydroshare/resource.py

```diff
@@ -722,16 +725,32 @@ def delete_resource_file(pk, filename_or_id, user):
 
     return filename_or_id
 
+def get_resource_doi(res_id, pending):
+    doi_str = "http://dx.doi.org/10.4211/hs.{shortkey}".format(shortkey=res_id)
+    if pending:
+        return "{doi}pending".format(doi=doi_str)
+    else:
+        return doi_str
+
+
+def get_activated_doi(doi):
+    idx = doi.find('pending')
+    if idx >= 0:
+        return doi[:idx]
+    else:
+        return doi
 
-def publish_resource(pk):
+def publish_resource(user, pk):
```

Comment:
A user can not do anything except request publication.  If their metadata is somehow deficient they should be told so, but beyond that they should just get a message that the DOI is pending assignment and then we should take care of this without further user intervention.  I think that if there is a transient network failure, HydroShare should re-queue the issue for some period (e.g. 3 hours) later and keep trying until successful.  If we get repeated failures then there is something systemic wrong that we need to address as developers.  I do not think we should be asking the user to try again except in extreme circumstances.  support@hydroshare should be automatically notified of such failures.